azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/Region.java

```diff
@@ -156,6 +156,20 @@ public static Region fromName(String name) {
     }
 
     /**
+     * Parses a region label into a Region object and creates a new Region instance if not found among the existing ones.
+     * @param label a region label
+     * @return the parsed or created region
+     */
+    public static Region fromLabel(String label) {
+        Region region = VALUES_BY_LABEL.get(label.toLowerCase());
+        if (region != null) {
+            return region;
+        } else {
+            return Region.create(label, label);
+        }
+    }
+
```

Comment:
interesting point about the spacing and casing as being ignored by the server per the spec, I didn't know that. That means that indeed the Region logic could be simplified and fromName() can safely treat labels as names (basically remove spaces, because we already ignore the casing). 
So let me remove .fromLabel() and instead make .fromName() support this.

We still need .label() as a read only property because it is giving the API consumer valuable information -- how to show this in their UI.